test/suites/network/multi-nic.robot

```diff
@@ -31,15 +34,66 @@ Verify MicroShift Runs On Both NICs
     # Wait for MicroShift API readiness and run verification
     Wait For MicroShift
     Verify Hello MicroShift LB
-    Verify Hello MicroShift NodePort    ${USHIFT_HOST_IP1}    ${USHIFT_HOST_IP2}
+    Verify Hello MicroShift NodePort    ${USHIFT_HOST_IP1}
+    Verify Hello MicroShift NodePort    ${USHIFT_HOST_IP2}
+    IP Should Be Present In External Certificate    ${USHIFT_HOST_IP1}
+    IP Should Be Present In External Certificate    ${USHIFT_HOST_IP2}
+
+Verify MicroShift Runs Only On Primary NIC
+    [Documentation]    Verify MicroShift can run only on the primary NIC. The
+    ...    node IP will be taken from this NIC by default. When disabling the
+    ...    secondary NIC nothing will happen in MicroShift, as the IP did not
+    ...    change. A restart is forced so that MicroShift picks up the new
+    ...    configuration (without the secondary IP) and regenerates the
+    ...    certificates, which will be lacking the IP from secondary NIC.
+    [Setup]    Save Default MicroShift Config
+
+    Configure Subject Alternative Name    ${USHIFT_HOST_IP1}
 
-Verify MicroShift Runs On First NIC
-    [Documentation]    Verify MicroShift can run on the first NIC
-    Verify MicroShift On One NIC    ${USHIFT_HOST_IP1}    ${NIC2_NAME}    ${USHIFT_HOST_IP1}    ${EMPTY}
+    ${cur_pid}=    MicroShift Process ID
+
+    Login Switch To IP    ${USHIFT_HOST_IP1}
+    Disable Interface    ${NIC2_NAME}
+
+    Restart MicroShift
+
+    Verify MicroShift On Single NIC    ${USHIFT_HOST_IP1}    ${USHIFT_HOST_IP2}
+
+    # Rebooting MicroShift host restores the network configuration
+    [Teardown]    Run Keywords
+    ...    Restore Default MicroShift Config
+    ...    Reboot MicroShift Host
+    ...    Login Switch To IP1
+    ...    Wait For Healthy System
 
-Verify MicroShift Runs On Second NIC
-    [Documentation]    Verify MicroShift can run on the second NIC
-    Verify MicroShift On One NIC    ${USHIFT_HOST_IP2}    ${NIC1_NAME}    ${EMPTY}    ${USHIFT_HOST_IP2}
+Verify MicroShift Runs Only On Secondary NIC
+    [Documentation]    Verify MicroShift can run only on the secondary NIC. The
+    ...    node IP will change when disabling the primary interface, triggering
+    ...    an automatic restart of the service. After restarting, the node IP will
+    ...    be that of the secondary NIC, and certificates will be updated according
+    ...    to the new configuration (which includes only the secondary IP).
+    [Setup]    Save Default MicroShift Config
+
+    Configure Subject Alternative Name    ${USHIFT_HOST_IP2}
+
+    ${cur_pid}=    MicroShift Process ID
+
+    Login Switch To IP    ${USHIFT_HOST_IP2}
+    Disable Interface    ${NIC1_NAME}
+
+    Wait Until MicroShift Process ID Changes    ${cur_pid}
+    Wait For MicroShift Service
+    Setup Kubeconfig
+    Wait For MicroShift
```

Comment:
Nit: we could probaby do following
```suggestion
    Wait Until MicroShift Process ID Changes    ${cur_pid}
    Wait For MicroShift
    Setup Kubeconfig
```

We could extract them into another kw: `Wait For Automatic Restart Of MicroShift`

Reply:
Thought of this but `Wait For MicroShift` makes use of the kubeconfig, so you need to set it up before that keyword (server IP is changing upon restart). Thats the reasoning behind the current order.
About the keyword, its only used here, is it worth of a new one for a single test?